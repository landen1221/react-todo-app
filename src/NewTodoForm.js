import React, { useState } from "react";

const NewTodoForm = ({ addTask }) => {
  const INITIAL_STATE = { todo: "" };

  const [formData, setFormData] = useState(INITIAL_STATE);

  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData((formData) => ({
      ...formData,
      [name]: value,
    }));
  };

  const handleSubmit = (e) => {
    e.preventDefault();

    addTask({ ...formData });
    setFormData(INITIAL_STATE);
  };

  return (
    <div>
      <h3>This is the form to add a task!</h3>
      <form onSubmit={handleSubmit}>
        <label htmlFor="todo">New Todo: </label>
        <input
          id="todo"
          type="text"
          name="todo"
          placeholder="ie: Walk the dog"
          value={formData.todo}
          onChange={handleChange}
        />
        <button>Add Task</button>
      </form>
    </div>
  );
};

export default NewTodoForm;
