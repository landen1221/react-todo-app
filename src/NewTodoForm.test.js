import { render } from '@testing-library/react';
import NewTodoForm from './NewTodoForm'

test('renders learn react link', () => {
    render(<NewTodoForm />);
});
  
it('should match snapshot', () => {
    const {asFragment} = render(<NewTodoForm />);
    expect(asFragment()).toMatchSnapshot();
})