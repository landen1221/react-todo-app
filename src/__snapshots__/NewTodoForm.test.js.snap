// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      This is the form to add a task!
    </h3>
    <form>
      <label
        for="todo"
      >
        New Todo: 
      </label>
      <input
        id="todo"
        name="todo"
        placeholder="ie: Walk the dog"
        type="text"
        value=""
      />
      <button>
        Add Task
      </button>
    </form>
  </div>
</DocumentFragment>
`;
